morphocluster/frontend/src/views/Files.vue

```diff
@@ -0,0 +1,167 @@
+                        }"><i class="mdi mdi-folder" />  {{ data.item.name }}</router-link>
+                        <router-link v-if="data.item.type === 'file'" :to="{
+                            name: 'file',
+                            params: { file_name: data.item.name, file_path: data.item.path },
+                        }"><i class="mdi mdi-file" />  {{ data.item.name}} </router-link>
+                    </template>
+                </b-table>
+            </div>
+        </div>
+        <div class="container mt-4">
+            <div class="dropzone" @dragover.prevent @dragenter.prevent @dragleave.prevent @drop="handleDrop">
+                Ziehe Dateien hierhin oder klicke, um Dateien auszuwählen
+            </div>
+            <input type="file" id="fileInput" style="display: none" @change="handleFileUpload" multiple />
+            <div class="container mt-4 text-center">
+                <button class="btn btn-primary" @click="openFileInput">Dateien auswählen</button>
+            </div>
+
+        </div>
+    </div>
+
+</template>
+
+<script>
+import "@mdi/font/css/materialdesignicons.css";
+import axios from "axios";
+import "bootstrap/dist/css/bootstrap.min.css";
+import "bootstrap";
+import { uploadFile } from "../helpers/api.js";
+
+export default {
+    name: "FilesView",
+    props: { file_path: String },
+    components: {},
+    response: "",
+    data() {
+        return {
+            fields: [
+                { key: "name", sortable: true },
+                "last_modified",
+            ],
+            files: [],
+            alerts: [],
+        };
+    },
+    created() {
+        this.initialize();
+    },
+    watch: {
+        $route: "initialize",
+    },
+    methods: {
+        openFileInput() {
+            document.getElementById("fileInput").click();
+        },
+        async handleFileUpload(event) {
+            const selectedFiles = event.target.files;
+
+            for (let i = 0; i < selectedFiles.length; i++) {
+                const file = selectedFiles[i];
+
+                try {
+                    const response = await uploadFile(file, this.file_path);
+                    console.log("Data upload successfull", response.message);
+                    window.location.reload();
+                } catch (error) {
+                    console.error("Data upload failed", error.message);
+                }
+            }
+        },
+        async initialize() {
+            try {
+                const response = await axios.get(`/api/files/${this.file_path}`);
+                this.files = response.data;
+            } catch (error) {
+                console.error(error);
+                this.alerts.unshift({
+                    message: error.message,
+                    variant: "danger",
+                });
+            }
+        },
+        // Funktion zum Verarbeiten des Drag-and-Drop-Ereignisses
+        async handleDrop(event) {
```

Comment:
handleDrop and handleFileUpload are virtually identical. Refactor common functionality into a method that is called by both.